src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -493,17 +493,19 @@ public void actionPerformed(final ActionEvent e) {
             }
             chooserScrollPane = new JScrollPane(chooser);
             final Dimension screenResolution = Toolkit.getDefaultToolkit().getScreenSize();
-            int availHeight = screenResolution.height - 80;
+            final int availHeight = screenResolution.height - 130;
             final int availWidth = screenResolution.width - 30;
-            availHeight -= 50;
+            chooserScrollPane.setBorder(BorderFactory.createEmptyBorder(1, 1, 1, 1));
+            final Dimension size = chooserScrollPane.getPreferredSize();
             chooserScrollPane.setPreferredSize(new Dimension(
-                (chooserScrollPane.getPreferredSize().width > availWidth ? availWidth
-                    : (chooserScrollPane.getPreferredSize().height > availHeight
-                        ? chooserScrollPane.getPreferredSize().width + 22
-                        : chooserScrollPane.getPreferredSize().width)),
-                (chooserScrollPane.getPreferredSize().height > availHeight ? availHeight
-                    : chooserScrollPane.getPreferredSize().height)));
-            chooserScrollPane.setBorder(new LineBorder(chooserScrollPane.getBackground()));
+                size.width > availWidth
+                    ? availWidth
+                    : size.width + (size.height > availHeight
+                        ? chooserScrollPane.getVerticalScrollBar().getPreferredSize().width
+                        : 0),
```

Comment:
Not 100% accurate, but far more readable, and perhaps more correct:
```java
Math.min(availWidth,
  size.width + (size.height > availHeight
    ? chooserScrollPane.getVerticalScrollBar().getPreferredSize().width
    : 0))
```

Reply:
Good call to replace one level of conditional operators with `Math#min()`. 0507e54a1